Branch-SDK/src/io/branch/referral/ServerRequestQueue.java

```diff
@@ -154,14 +155,16 @@ void enqueue(ServerRequest request) {
      */
     ServerRequest dequeue() {
         ServerRequest req = null;
-        try {
-            req = queue.remove(0);
-            persist();
-        } catch (IndexOutOfBoundsException | NoSuchElementException ignored) {
+        synchronized (reqQueueLockObject) {
+            try {
+                req = queue.remove(0);
```

Comment:
The extraneous try/catches are removable if you return from the public functions that already exist with the try catches.  In this case it would be removeAt(0);

That way you're not duplicating code/error handling.  Not a deal breaker for the PR though and could be moved to a new one considering it isn't related to the actual fix this is addressing